common/src/main/java/org/apache/drill/common/config/NestedConfig.java

```diff
@@ -111,6 +111,14 @@ public String getString(String path) {
     return c.getString(path);
   }
 
+  public String getString(String path, String defaultValue) {
+    String value = hasPath(path) ? getString(path) : null;
+    if (value == null || (value = value.trim()).isEmpty()) {
+      return defaultValue;
+    }
+    return value;
```

Comment:
```StringUtils.defaultIfBlank``` changes method logic, since in an original method was: 
```java
  private String getConfigParamWithDefault(String name, String defaultValue) {
    String value = "";
    if (config.hasPath(name)) {
      value = config.getString(name);
    }
    if (value.isEmpty()) {
      value = defaultValue;
    }
    value = value.trim();
    return value;
  }
```
Although I don't think that trimming was actually necessary, so I'll try your suggestion.

Reply:
There is no need for this method at all, default values can be set directly in drill-module.conf.